transport/transport.go

```diff
@@ -330,12 +330,20 @@ func NewServerTransport(protocol string, conn net.Conn, maxStreams uint32, authI
 	return newHTTP2Server(conn, maxStreams, authInfo)
 }
 
+// DefaultDialerFn is the default dialer.
+func DefaultDialerFn(addr string, timeout time.Duration, cancel <-chan struct{}) (net.Conn, error) {
```

Comment:
`DefaultDialerFn` is used outside the tests and depends on `MakeDefaultDialerFn`, so moving these functions to the tests is not possible. Copying them is possible, but is that better?

As I said, if you want to avoid external users depending on `transport`, you can move it to `internal/transport`. See https://golang.org/s/go14internal.